src/app/shared/components/cart-tooltip/cart-tooltip.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, debounceTime, Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartTooltipService {
+  isVisible$ = new BehaviorSubject(false).pipe(debounceTime(500));
+
+  constructor() {}
+
+  public open() {
+    (this.isVisible$ as Subject<boolean>).next(true);
```

Comment:
You don't need to specify the type "Subject", because BehaviorSubject is not a Subject, but only a difference from it, it also has a method "next()".

Reply:
I did it because on the line 8 was used pipe with debounceTime.
It turn Subject in Observable and i got "TS2339: Property 'next' does not exist on type 'Observable '" error

Reply:
Usually they don't do this in a variable declaration, such manipulations happen at the moment of subscription to a Subject, I suggest to make a method getVisibleObs() which will return "Observable", you can also add "debounceTime" in this method.